projects/noloan/app/src/main/java/hasadna/noloan/SpamRecyclerAdapter.java

```diff
@@ -29,26 +31,80 @@
     extends RecyclerView.Adapter<SpamRecyclerAdapter.RecyclerViewHolder> {
   private static final String TAG = "SpamRecyclerAdapter";
 
+  ArrayList<SmsMessage> inboxSpam;
+
   public SpamRecyclerAdapter() {
     Handler handler = new Handler(Looper.myLooper());
+    inboxSpam = new ArrayList<>();
+
+    // Fetch spams from DB that are in user's inbox
+    for (int i = 0; i < SmsMessages.get().getInboxMessages().size(); i++) {
+      for (SmsMessage sms : SmsMessages.get().getDbMessages()) {
+        if (SmsMessages.get().getInboxMessages().get(i).getBody().contentEquals(sms.getBody())
+            && SmsMessages.get()
+                .getInboxMessages()
+                .get(i)
+                .getSender()
+                .contentEquals(sms.getSender())) {
+          // Add message from the DB - keep received date of user's message
+          inboxSpam.add(
+              sms.toBuilder()
+                  .setReceivedAt(SmsMessages.get().getInboxMessages().get(i).getReceivedAt())
+                  .build());
+        }
+      }
+    }
 
-    // Listen to db smsMessages
+    // Listen to db smsMessages. Update just relevant messages that are in the inbox as well.
+    // Keep the received date of the user's message.
     SmsMessages.get()
         .setMessagesListener(
             new SmsMessages.MessagesListener() {
               @Override
               public void messageAdded(SmsMessage newMessage) {
-                handler.post(() -> notifyItemInserted(SmsMessages.get().getDbMessages().size()));
+                int inboxIndex =
+                    SmsMessages.get()
+                        .searchMessage(newMessage, SmsMessages.get().getInboxMessages());
```

Comment:
Cache the 'InboxMessages' since it is used in multiple places.
Same for 'DbMessages'.

Reply:
I got the "hang" of what you were saying about having some `SmsMessages` function statics, and that it's easier to use (instead of `.get()` each time) as I wrote this...

I didn't want to push these changes (concerning SmsMessages static functions) in this PR, so left it like that for now.
Lets raise this idea again in the channel...

Reply:
getInboxMessages and getDbMessages cannot be static they use the class fields.

just before the call for setMessagesListener (bad name BTW, you are adding not setting) save thous into a variable and use it instead.
calling the same function more then once is not a good thing, It is maybe OK in our case because it is just a get function, but generally it is something you want to avoid.

Plus it make the code longer then it is actually need to be and harder to read.